x-pack/solutions/search/plugins/search_query_rules/public/plugin.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, Plugin, AppMountParameters, CoreStart } from '@kbn/core/public';
+import { PLUGIN_ID, PLUGIN_NAME, PLUGIN_TITLE } from '../common';
+import {
+  AppPluginSetupDependencies,
+  AppPluginStartDependencies,
+  SearchQueryRulesPluginSetup,
+  SearchQueryRulesPluginStart,
+} from './types';
+import { QUERY_RULES_UI_FLAG } from '../common/ui_flags';
+import { docLinks } from '../common/doc_links';
+import { PLUGIN_ROUTE_ROOT } from '../common/api_routes';
+
+export class QueryRulesPlugin
+  implements Plugin<SearchQueryRulesPluginSetup, SearchQueryRulesPluginStart>
+{
+  constructor() {}
+
+  public setup(
+    core: CoreSetup<AppPluginStartDependencies, SearchQueryRulesPluginStart>,
+    _: AppPluginSetupDependencies
+  ): SearchQueryRulesPluginSetup {
+    if (!core.settings.client.get<boolean>(QUERY_RULES_UI_FLAG, false)) {
+      return {};
+    }
+    core.application.register({
+      id: PLUGIN_ID,
+      appRoute: PLUGIN_ROUTE_ROOT,
+      title: PLUGIN_TITLE,
+      deepLinks: [
+        {
```

Comment:
I don't think we need this deeplink since one will automatically be created for the app. you can just add `'globalSearch'` `visibleIn` array on the app object on line 59. right?